### PR TITLE
SONAR-9028 drop table authors

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/DbVersion65.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v65;
+
+import org.sonar.server.platform.db.migration.step.MigrationStepRegistry;
+import org.sonar.server.platform.db.migration.version.DbVersion;
+import org.sonar.server.platform.db.migration.version.v64.AddTagsToProjects;
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/DbVersion65.java]: MINOR Remove this unused import 'org.sonar.server.platform.db.migration.version.v64.AddTagsToProjects'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select distinct gr.resource_id from group_roles gr where" +
+        " gr.resource_id is not null" +
+        " and not exists (select id from projects p where p.id = gr.resource_id)");
+    massUpdate.rowPluralName("rows without component");
+    massUpdate.update("delete from group_roles where resource_id = ?");
+    massUpdate.execute(CleanOrphanRowsInGroupRoles::handle);
+  }
+
+  private void deleteRowsForNonRootComponent(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java]: MAJOR Make "deleteRowsForNonRootComponent" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    super(db);
+  }
+
+  @Override
+  protected void execute(Context context) throws SQLException {
+    deleteRowsWithoutComponent(context);
+    deleteRowsForNonRootComponent(context);
+  }
+
+  private void deleteRowsWithoutComponent(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java]: MAJOR Make "deleteRowsWithoutComponent" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select distinct ur.resource_id from user_roles ur where" +
+      " ur.resource_id is not null" +
+      " and not exists (select id from projects p where p.id = ur.resource_id)");
+    massUpdate.rowPluralName("rows without component");
+    massUpdate.update("delete from user_roles where resource_id = ?");
+    massUpdate.execute(CleanOrphanRowsInUserRoles::handle);
+  }
+
+  private void deleteRowsForNonRootComponent(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java]: MAJOR Make "deleteRowsForNonRootComponent" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    super(db);
+  }
+
+  @Override
+  protected void execute(Context context) throws SQLException {
+    deleteRowsWithoutComponent(context);
+    deleteRowsForNonRootComponent(context);
+  }
+
+  private void deleteRowsWithoutComponent(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java]: MAJOR Make "deleteRowsWithoutComponent" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  private void deleteRowsForNonRootComponent(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select distinct ur.resource_id from user_roles ur" +
+      " inner join projects p on p.id = ur.resource_id" +
+      " where" +
+      " p.scope <> ?" +
+      " or (p.qualifier <> ? and p.qualifier <> ?)")
+      .setString(1, "PRJ")
+      .setString(2, "TRK")
+      .setString(3, "VW");
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInUserRoles.java]: "SVW" can have direct permissions. It should be added to the list of accepted qualifiers.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  private void deleteRowsForNonRootComponent(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select distinct gr.resource_id from group_roles gr" +
+        " inner join projects p on p.id = gr.resource_id" +
+        " where" +
+        " p.scope <> ?" +
+        " or (p.qualifier <> ? and p.qualifier <> ?)")
+        .setString(1, "PRJ")
+        .setString(2, "TRK")
+        .setString(3, "VW");
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java]: "SVW" can have direct permissions. It should be added to the list of accepted qualifiers.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  protected void execute(Context context) throws SQLException {
+    deleteRowsWithoutComponent(context);
+    deleteRowsForNonRootComponent(context);
+  }
+
+  private void deleteRowsWithoutComponent(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select distinct gr.resource_id from group_roles gr where" +
+        " gr.resource_id is not null" +
+        " and not exists (select id from projects p where p.id = gr.resource_id)");
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInGroupRoles.java]: Couldn't it filter on `projects.enabled=true` too ? If yes then it should also be added to other SQL requests.

**Comment by simonbrandhof** [comment]
Last commit is in bad branch, otherwise LGTM.

**Pull request merged**